### PR TITLE
SageMaker ProcessingJob Controller with E2E tests

### PR DESCRIPTION
**Pull request by RedbackThomson**
Issue #, if available:

Description of changes:
- Add `ProcessingJob` support
- Add `kmeans` processing E2E test

Minor changes:
- Add pytest-xdist to parallelise tests
- Cleanup helm test namespace

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by RedbackThomson** [comment]
Marked with `do-not-merge` until SM team has given two approvals

File: test/e2e/common/k8s.py
Comment (mbaijal) [inline test/e2e/common/k8s.py:147]: 👍🏻

File: test/e2e/sagemaker/resources/kmeans_processingjob.yaml
Comment (mbaijal) [inline test/e2e/sagemaker/resources/kmeans_processingjob.yaml:17]: Were all of these fields always required in the spec file (with the older operators I mean ?)

**Review by mbaijal — APPROVED** [review]

File: test/e2e/sagemaker/resources/kmeans_processingjob.yaml
Comment (RedbackThomson) [reply test/e2e/sagemaker/resources/kmeans_processingjob.yaml:17]: I'm not sure. I checked the file history and this particular line has been in there since day 1. I might actually experiment with removing this, I think this is the only unnecessary line.

**Comment by RedbackThomson** [comment]
Want to add to all SageMaker tests that it deletes the k8s resources after finishing. That way we aren't polluting the default namespaces.

**Comment by surajkota** [comment]
had a quick look, lgtm

**Pull request merged**
